### PR TITLE
Closes #221: Feature request - Provide CSV files

### LINKED ISSUE #221: Feature Request:  Provide CSV files
### Feature Idea

Python becomes more and more popular in the space of Data Science in Pharma and clinical data transformation and analytics. 

It would be great to be able to use pharmaversesdtm test data in Python development.

If pharmaversesdtm would provide the data sets in addition to `*.rda` files in /data also as `*.csv` files e.g. in the same folder `/data/*.csv`, then these csv files could be easily used in Python directly from Github.

```python
import pandas as pd

url = "https://raw.githubusercontent.com/pharmaverse/pharmaversesdtm/refs/heads/main/data/ae.csv"
df = pd.read_csv(url)
```

Guess and hope the csv creation from RDA files at the end of pharmaversesdtm process is a small extention of the pharmaversesdtm test data creation process. 

### Relevant Input

_No response_

### Relevant Output

All `*.rda` files currently in `/data` should be available also as `*.csv` files

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by Gero1999**
> Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

>Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

___

# Implementation description

This pull request introduces the automated export of SDTM datasets to CSV format and adds two example CSV files to the repository. The changes improve reproducibility and make example datasets more accessible for external use or testing.

- The data generation script (`data-raw/create_sdtms_data.R`) now saves a CSV version of each dataset to the `inst/extdata/` directory, making the datasets directly available for non R-programmers. 

- The `.Rbuildignore` file is updated to exclude all CSV files in `inst/extdata/` from the R package build, ensuring these files are not included in the built package by default for CRAN submissions.

___

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build pharmaversesdtm site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by hski-bayer** [comment]
There is pandas.read_json(URL). But the Json need to follow certain structures. There is 'split', 'records' and 'index'. 

See https://pandas.pydata.org/docs/reference/api/pandas.read_json.html

But I don't see how meta data could be preserved resp utilized.

**Comment by hski-bayer** [comment]
Okay, agreed. Then please go back to the original proposal to create CSV, because that would nicely work with Python pandas.read_csv and then pharmaverse datasets can be used in Python.

**Comment by bundfussr** [comment]
> Yep right, we can provide CSV then. But I think it would still be worthy to consider the idea of creating another issue to provide datasetJSON files as well. What do you think @manciniedoardo @bundfussr ?
> 
> In the meantime, I will create [a package](https://github.com/Gero1999/biojson) to allow the Python community to deal with `datasetJSON` as well as any other bio/pharma JSON standard. It might take me a bit to make something solid and publish, but I think it might be worthy

Yes, I think it makes sense to create CSVs as a temporary solution. Once the python package is available we can replace the CSV files with dataset-JSON files and then also provide dataset-JSON files in pharmaverseadam.

File: README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # pharmaversesdtm <img src="man/figures/logo.png" align="right" width="200" style="margin-left:50px;" alt="pharmaverse sdtm hex"/>
 
+> <sup>Interactive data exploration: <a href="https://pharmaverse.github.io/pharmaversesdtm/articles/preview-sdtm.html">Preview SDTM vignette</a></sup>
```
Comment (manciniedoardo) [inline README.md]: Should this be part of this PR? @Lina2689

File: README.md [hunk 2]
```diff
@@ -31,10 +33,17 @@ if (!requireNamespace("remotes", quietly = TRUE)) {
...
+
+## Data {#data}
+
+### Data Sources
 
 Some test datasets have been sourced from the [CDISC pilot project](https://github.com/cdisc-org/sdtm-adam-pilot-project), while other datasets have been constructed ad-hoc by the {admiral} team. Please check the [Reference page](https://pharmaverse.github.io/pharmaversesdtm/reference/index.html) for detailed information regarding the source of specific datasets.
 
+### Data Formats
+
+Each dataset is provided in both RDA and CSV formats. The `.rda` files are used within the R package and saved in the [data](https://github.com/pharmaverse/pharmaversesdtm/tree/main/data) folder. The `.csv` files are saved in the [inst/extdata/](https://github.com/pharmaverse/pharmaversesdtm/tree/main/inst/extdata) folder.
```
Comment (Lina2689) [inline README.md]:
```suggestion
Each dataset is provided in both RDA and CSV formats. The `.rda` files are used within the R package and saved in the [data](https://github.com/pharmaverse/pharmaversesdtm/tree/main/data) folder, while the `.csv` files are saved in the [inst/extdata/](https://github.com/pharmaverse/pharmaversesdtm/tree/main/inst/extdata) folder.
```

File: README.md [hunk 3]
```diff
@@ -78,7 +87,7 @@ This metadata drives the automated documentation process, and the file is read b
 
 ### Adding New SDTM Datasets
 
--   Create a program in the `data-raw/` folder, named `<name>.R`, where `<name>` should follow the [naming convention](#naming), to generate the test data and output `<name>.rda` to the `data/` folder.
+-   Create a program in the `data-raw/` folder, named `<name>.R`, where `<name>` should follow the [naming convention](#naming), to generate the test data and output `<name>.rda` to the `data/` folder, as well as `<name>.csv` to the `inst/extdata` folder.
```
Comment (Lina2689) [inline README.md]:
```suggestion
-   Create a program in the `data-raw/` folder, named `<name>.R`, where `<name>` should follow the [naming convention](#naming), to generate the test data and output `<name>.rda` to the `data/` folder, and the `<name>.csv` to the `inst/extdata` folder.
```
